x/stake/handler.go

```diff
@@ -217,81 +181,80 @@ func handleMsgUnbond(ctx sdk.Context, msg MsgUnbond, k Keeper) sdk.Result {
 		}
 	} else {
 		var err sdk.Error
-		shares, err = sdk.NewRatFromDecimal(msg.Shares)
+		delShares, err = sdk.NewRatFromDecimal(msg.Shares)
 		if err != nil {
 			return err.Result()
 		}
-		if bond.Shares.LT(shares) {
+		if bond.Shares.LT(delShares) {
 			return ErrNotEnoughBondShares(k.codespace, msg.Shares).Result()
 		}
 	}
 
-	// get candidate
-	candidate, found := k.GetCandidate(ctx, msg.CandidateAddr)
+	// get validator
+	validator, found := k.GetValidator(ctx, msg.ValidatorAddr)
 	if !found {
-		return ErrNoCandidateForAddress(k.codespace).Result()
+		return ErrNoValidatorForAddress(k.codespace).Result()
 	}
 
 	if ctx.IsCheckTx() {
-		return sdk.Result{
-			GasUsed: GasUnbond,
-		}
+		return sdk.Result{}
 	}
 
 	// retrieve the amount of bonds to remove (TODO remove redundancy already serialized)
 	if msg.Shares == "MAX" {
```

Comment:
interesting.

Reply:
talking with some folks in the validator group - people also want the ability to unbond a % of the bond as opposed to an amount of shares